File: 06-3-BoardOfMessagesApp/app/src/main/java/com/example/boardofmessagesapp/repository/retrofit/BoardInterface.java
package com.example.boardofmessagesapp.repository.retrofit;

import com.example.boardofmessagesapp.repository.Board;
import com.example.boardofmessagesapp.repository.GeneralResponse;
import com.example.boardofmessagesapp.repository.Message;

import java.util.List;

import retrofit2.Call;
import retrofit2.http.Body;
import retrofit2.http.DELETE;
import retrofit2.http.GET;
import retrofit2.http.POST;
import retrofit2.http.Path;

public interface BoardInterface {

    @GET("board")
    public Call<List<Board>> getBoards();

    @POST("board")
    public Call<Board> addBoard(@Body Board board);

    @GET("board/{iid}")
    public Call<Board> getBoard(@Path("iid") long iid);

    @DELETE("board/{iid}")
    public Call<GeneralResponse> deleteBoard(@Path("iid") long iid);

    @POST("board/{iid}/messages")
    public Call<GeneralResponse> postMessage(@Path("iid") long iid, @Body Message message);

    // only for testing
    @GET("reset")
    public Call<GeneralResponse> reset();

    @GET("board/{iid}/messages")
    public Call<List<Message>> getMessages(@Path("iid") long iid);
}

File: 06-3-BoardOfMessageApp_kt2/app/src/main/java/com/example/boardofmessageapp_kt2/repository/retrofit/BoardInterface.kt
package com.example.boardofmessageapp_kt2.repository.retrofit

import com.example.boardofmessageapp_kt2.repository.common.Board
import com.example.boardofmessageapp_kt2.repository.common.GeneralResponse
import com.example.boardofmessageapp_kt2.repository.common.Message
import retrofit2.http.Body
import retrofit2.http.DELETE
import retrofit2.http.GET
import retrofit2.http.POST
import retrofit2.http.Path


interface BoardInterface {

    @GET("board")
    suspend fun getBoards() : List<Board>

    @POST
    suspend fun addBoard(@Body board: Board): Board

    @GET("board/{iid}")
    suspend fun getBoard(@Path("iid") iid: Long): Board

    @DELETE("board/{iid}")
    suspend fun deleteBoard(@Path("iid") iid: Long): GeneralResponse

    @POST("board/{iid}/messages")
    suspend fun postMessage(@Path("iid") iid: Long, @Body message: Message): GeneralResponse

    // only for testing
    @GET("reset")
    suspend fun reset(): GeneralResponse

    @GET("board/{iid}/messages")
    suspend fun getMessages(@Path("iid") iid: Long): List<Message>

}